src/components/transactions/TxDetails/TxData/ExecTransaction/index.tsx

```diff
@@ -0,0 +1,106 @@
+import { Safe__factory } from '@/types/contracts'
+import { Skeleton, Stack, SvgIcon, Typography } from '@mui/material'
+import { getConfirmationView, type TransactionData } from '@safe-global/safe-gateway-typescript-sdk'
+import ErrorMessage from '@/components/tx/ErrorMessage'
+import { MethodDetails } from '../DecodedData/MethodDetails'
+import MethodCall from '../DecodedData/MethodCall'
+import DecodedTx, { Divider } from '@/components/tx/DecodedTx'
+import Link from 'next/link'
+import { useCurrentChain } from '@/hooks/useChains'
+import { AppRoutes } from '@/config/routes'
+import { useMemo } from 'react'
+import type { SafeTransaction } from '@safe-global/safe-core-sdk-types'
+import useAsync from '@/hooks/useAsync'
+import NestedTransactionIcon from '@/public/images/transactions/nestedTx.svg'
+import ExternalLink from '@/components/common/ExternalLink'
+
+const safeInterface = Safe__factory.createInterface()
+
+const extractTransactionData = (data: string | undefined): SafeTransaction | undefined => {
+  const params = data ? safeInterface.decodeFunctionData('execTransaction', data) : undefined
+  if (!params || params.length !== 10) {
+    return
+  }
+
+  return {
+    addSignature: () => {},
+    encodedSignatures: () => params[9],
+    getSignature: () => undefined,
+    data: {
+      to: params[0],
+      value: params[1],
+      data: params[2],
+      operation: params[3] as number,
```

Comment:
We don't need to cast this as the type is inherently `any`. (I'm quite surprised it's missing the types.)